core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -220,24 +225,40 @@ trait SubmissionComponent {
               .map(_.id)
           )
 
-          DBIO.sequenceOption(
+          val entityIdAction = DBIO.sequenceOption(
             submission.submissionEntity.map(loadSubmissionEntityId(workspaceContext.workspaceIdAsUUID, _))
-          ) flatMap { entityId =>
-            configIdAction flatMap { configId =>
-              if (configId.isEmpty) {
-                throw new RawlsExceptionWithErrorReport(
-                  ErrorReport(
-                    StatusCodes.BadRequest,
-                    s"Can't find this submission's method config ${submission.methodConfigurationNamespace}/${submission.methodConfigurationName}."
-                  )
-                )
-              }
-              submissionStatusCounter(submission.status).countDBResult {
-                submissionQuery += marshalSubmission(workspaceContext.workspaceIdAsUUID,
-                                                     submission,
-                                                     entityId,
-                                                     configId.get
+          )
+
+          val entitiesIdsAction = submission.submissionEntities match {
+            case None =>
+              DBIO.successful(None)
+            case Some(entities) =>
+              DBIO
+                .sequence(
+                  entities.map(loadSubmissionEntityId(workspaceContext.workspaceIdAsUUID, _))
                 )
+                .map(seq => if (seq.isEmpty) None else Some(seq))
+          }
+
+          entityIdAction flatMap { entityId =>
```

Comment:
There's probably a smarter way to do this, but essentially we have to make different queries here depending on whether we have a single or multiple entity ids.